### ISSUE TITLE
Prescription Logic change

### ISSUE THREAD

**Issue by AnnieMungai** [opensrp/opensrp-client-giz-malawi#471]
- [ ] The system not to allow you to dispense the treatment you haven't prescribed. When a user wants to dispense a list of prescribed drugs to appear from which a user will select to dispense. 

**The ask** 
On the lab form, have a list of medications prepopulated when prescribed. Name to be **Medication Prescribed.** 
Tapping on any of the medications will open the question; Medication dispensed? Users should be able to select yes or no. 
If yes is selected, the prescription details form opens up. If no is selected, the reason why the medication was not prescribed pops open. 

**Question for discussion**
- What happens if the prescriptions have not gotten to the pharmacy? Or data from the clinician is not yet synced and so the pharmacy form cannot be prepopulated? Leave the add medication option still intact? 
- Currently, selecting the medication is used to tally the reports. How will this work if we take this implementation?

**Comment by dubdabasoduba** [comment]
**Technical scoping**
**Technical implementation.** 
- [ ] Create a repeating group on the pharmacy form.  This form should have the following. 
     - [ ] The repeating group number of groups will be determined by the number of medications prescribed in the previous form. 
     - [ ] A medication multi-select widget. This widget should be auto-populated by each medication from the prescribed medication.
     - [ ] A radio button question asking whether the medication is available
- [ ] The special instructions & referral questions will still remain on the final form.
- [ ] The referral question will be on the main form.  The relevance for this question will be determined as follows:
     - [ ] This question should be shown only if any of the `medication available` questions in the repeating group is answered as no. 

**Concerns**
- [ ] Can the Multi-Select widget be used in a repeating group?
- [ ] Can the answers of a question in a repeating group affect the relevance of a question outside the repeating groups. 

**Next Steps**
- [ ] Communicate this option to the client then get sign off for the POC (Proof Of Concept)
- [ ] Create a POC (Proof Of Concept) of the above suggested technical implementation. The LOE for this implementation is added above. It's a bit large due to the R&D and potential Native Forms work in case the concerns above need to be implemented

**Questions**
- Does the `special instructions` question get answered for each medication in the prescription? 

@AnnieMungai @roywanyaga @hamza-vd Please review this.

**Comment by AnnieMungai** [comment]
@dubdabasoduba no, it is one question for the whole prescription.

Question: If for example someone has a prescription for 5 medications, and only 3 are available in the pharmacy, the pharmacist will have to remove the ones that are not available. Will this implementation cater to this?

**Comment by roywanyaga** [comment]
@AnnieMungai yes, this implementation will cater for your scenario. There will be a question under each medication specifying whether its available or not.
